src/domain/ledger/index.types.d.ts

```diff
@@ -217,7 +218,13 @@ interface ILedgerService {
     args: AddUsernameIntraledgerTxSendArgs,
   ): Promise<LedgerJournal | LedgerServiceError>
 
-  settlePendingLnPayments(paymentHash: PaymentHash): Promise<boolean | LedgerServiceError>
+  settlePendingLnPayments({
+    paymentHash,
+    paymentId,
+  }: {
+    paymentHash: PaymentHash
+    paymentId: PaymentId
+  }): Promise<boolean | LedgerServiceError>
```

Comment:
I guess `boolean` means `success` here? and this can be only be true? is it a pattern we have adopted? wouldn't it be better if it was `true` instead of `boolean` since it can't be false?

(I know it's not part of the PR so no need to fix it here, just trying to understand the new code modifications)

Reply:
That's usually right if the function can only return `true` or an error, but in this case that method returns `result.nModified > 0` which can also be false if no records are modified (I believe).

Reply:
in that case we want want to trigger an error instead of `false`? 

none of the place where `settlePendingLnPayments` is being called really check for the value of the boolean currently

Reply:
We potentially can. I didn't include this initially to reflect the original behaviour of this method [here](https://github.com/GaloyMoney/galoy/blob/aa00a0a2e55e147dbb2e4f0a6c6c95aaa57608e5/src/core/lightning/index.ts#L516) & [here](https://github.com/GaloyMoney/galoy/blob/aa00a0a2e55e147dbb2e4f0a6c6c95aaa57608e5/src/core/lightning/index.ts#L545) where we didn't do anything with the return (`true` or `false`), but where if an error happened the entire method would throw.

In the new code we still don't do anything with the return (`true`/`false`), but we return an error where before the method would have just thrown it. 

Do we want to change the behaviour of this function so that `false` is an error as well. I haven't tested for any cases as yet where that could potentially happens and what the implications could be.